### PR TITLE
Python: Remove MotorCommandCursor in import typing

### PR DESCRIPTION
**Pull request by Jibola**
### Motivation and Context
#4433 

### Description
* If `tornado` is not installed, importing `MotorCommandCursor` will fail due to the conditional import requirement. The fix is to import its parent class `AgnosticCommandCursor`

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Comment by Jibola** [comment]
cc: @shaneharvey, @blink1073 , @caseyclements, @noahstapp

File: python/semantic_kernel/connectors/memory/mongodb_atlas/mongodb_atlas_memory_store.py [hunk 1]
```diff
@@ -268,7 +268,7 @@ async def get_nearest_matches(
         if min_relevance_score:
             pipeline.append({"$match": {"$gte": ["$score", min_relevance_score]}})
 
-        cursor: MotorCommandCursor = self.database[collection_name].aggregate(pipeline)
+        cursor: core.AgnosticCommandCursor = self.database[collection_name].aggregate(pipeline)
```
Comment (ShaneHarvey) [inline python/semantic_kernel/connectors/memory/mongodb_atlas/mongodb_atlas_memory_store.py:271]: Is this type annotation even needed? What happens if we remove it?

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: python/semantic_kernel/connectors/memory/mongodb_atlas/mongodb_atlas_memory_store.py (hunk 1 above)
Comment (Jibola) [reply python/semantic_kernel/connectors/memory/mongodb_atlas/mongodb_atlas_memory_store.py:271]: Honestly, it should be perfectly fine if we just remove it outright.

**Review by ShaneHarvey — APPROVED** [review]

**Review by eavanvalkenburg — APPROVED** [review]

**Comment by markwallace-microsoft** [comment]
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 967 | 11 :zzz: | 0 :x: | 0 :fire: | 19.267s :stopwatch: |

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/84757f303d97071258a75477b41bde7327f23c27/README.md">Python Test Coverage</a><details><summary>Python Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td><b>TOTAL</b></td><td><b>4781</b></td><td><b>1051</b></td><td><b>78%</b></td><td>&nbsp;</td></tr></tbody></table><i>report-only-changed-files is enabled. No files were changed during this commit :)</i></details>

## Python Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 978 | 11 :zzz: | 0 :x: | 0 :fire: | 18.609s :stopwatch: |

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/a29048d32f675b2b45117d695e3200631a474188/README.md">Python Test Coverage</a><details><summary>Python Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td><b>TOTAL</b></td><td><b>4781</b></td><td><b>1051</b></td><td><b>78%</b></td><td>&nbsp;</td></tr></tbody></table><i>report-only-changed-files is enabled. No files were changed during this commit :)</i></details>

## Python Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 978 | 11 :zzz: | 0 :x: | 0 :fire: | 21.935s :stopwatch: |

**Comment by moonbox3** [comment]
> Hey @markwallace-microsoft is everything good to go here? :) I went ahead and updated the branch this morning.

For Python related PRs, please reach out to me or @eavanvalkenburg.

**Review by moonbox3 — APPROVED** [review]

**Pull request merged**